### PR TITLE
Remove old branches from branches.txt

### PR DESCRIPTION
**Pull request by CaseyHillers**
This is causing a few errors in the backend related to vacuum github commits not having access to these branches

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by CaseyHillers** [comment]
Friendly ping

File: app_dart/dev/branches.txt [hunk 1]
```diff
@@ -1,4 +1 @@
-flutter-2.2-candidate.10
-flutter-2.4-candidate.4
-experimental_skshaper
```
Comment (godofredoc) [inline app_dart/dev/branches.txt:3]: skshaper may still be used, do you know if there are new commits to that branch?

Comment (CaseyHillers) [reply app_dart/dev/branches.txt:3]: Latest commit is 2 weeks old - https://github.com/flutter/flutter/tree/experimental_skshaper and https://github.com/flutter/engine/tree/experimental_skshaper

\cc @jason-simmons 

Generally, these branches are broken and aren't showing anything regardless through Cocoon. Now that ci.yaml controls where we run checks on a branched PR, this file should no longer be necessary.

Comment (CaseyHillers) [reply app_dart/dev/branches.txt:3]: As another point, I talked to @christopherfujino offline and we discussed replacing this with a request for https://cs.opensource.google/flutter/infra/+/main:config/lib/release_branches/branches.json instead

**Review by godofredoc — APPROVED** [review]

**Pull request merged**